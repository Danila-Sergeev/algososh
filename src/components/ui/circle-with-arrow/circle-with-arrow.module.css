.arrow {
  margin-left: 15px;
  font-weight: 700;
  color: var(--default-color);
}

.container {
  display: flex;
  align-items: center;
  position: relative;
}

.containerAbove .circle {
  top: -70px;
}

.containerBelow .circle {
  bottom: -70px;
}
