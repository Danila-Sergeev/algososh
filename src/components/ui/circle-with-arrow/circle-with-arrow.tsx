import React from "react";
import { Circle } from "../circle/circle";
import { ElementStates } from "../../../types/element-states";
import styles from "./circle-with-arrow.module.css";

interface CircleWithArrowProps {
  state?: ElementStates;
  letter?: string;
  head?: string | React.ReactElement | null;
  index?: number;
  tail?: string | React.ReactElement | null;
  tailType?: "string" | "element";
  extraClass?: string;
  isSmall?: boolean;
  showArrow?: boolean; // добавлено для контроля показа стрелки
  isAbove?: boolean;
}

export const CircleWithArrow: React.FC<CircleWithArrowProps> = ({
  showArrow = false, // новый проп
  isAbove = false,

  ...props
}) => {
  const containerClass = isAbove
    ? styles.containerAbove
    : styles.containerBelow;

  return (
    <div className={`${styles.container} ${containerClass}`}>
      <Circle {...props} />
      {showArrow && (
        <div className={styles.arrow} data-testid="arrow">
          &gt;
        </div>
      )}
    </div>
  );
};
