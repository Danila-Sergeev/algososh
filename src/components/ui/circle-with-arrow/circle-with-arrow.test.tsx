import React from "react";
import { render } from "@testing-library/react";
import { CircleWithArrow } from "./circle-with-arrow"; // Измените путь к вашему компоненту
import styles from "./circle-with-arrow.module.css";

describe("CircleWithArrow Component", () => {
    it("рендерит элемент без стрелки", () => {
        const { asFragment, queryByTestId } = render(<CircleWithArrow />);
        expect(asFragment()).toMatchSnapshot();
        expect(queryByTestId("arrow")).toBeNull(); // убедимся, что стрелка не отображается
    });

    it("рендерит элемент со стрелкой", () => {
        const { asFragment, getByTestId } = render(
            <CircleWithArrow showArrow />
        );
        expect(asFragment()).toMatchSnapshot();
        expect(getByTestId("arrow")).toBeDefined(); // убедимся, что стрелка отображается
    });
});
