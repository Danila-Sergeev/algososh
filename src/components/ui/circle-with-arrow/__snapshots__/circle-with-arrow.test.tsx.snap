// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleWithArrow Component рендерит элемент без стрелки 1`] = `
<DocumentFragment>
  <div
    class="container containerBelow"
  >
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
        data-test-id="circleHead"
      />
      <div
        class="circle   default"
        data-test-id="circle"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-test-id="circleTail"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`CircleWithArrow Component рендерит элемент со стрелкой 1`] = `
<DocumentFragment>
  <div
    class="container containerBelow"
  >
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
        data-test-id="circleHead"
      />
      <div
        class="circle   default"
        data-test-id="circle"
      >
        <p
          class="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-test-id="circleTail"
      />
    </div>
    <div
      class="arrow"
      data-testid="arrow"
    >
      &gt;
    </div>
  </div>
</DocumentFragment>
`;
